src/TraverseTracks.cpp

```diff
@@ -0,0 +1,153 @@
+#include "TraverseTracks.h"
+
+/**
+ * @brief Constructor for the TravseSegments class assigns the TrackGenerator
+ *        and pulls relevant information from it.
+ */
+TraverseTracks::TraverseTracks(TrackGenerator* track_generator) {
+
+  /* Save the track generator */
+  _track_generator = track_generator;
+
+  /* Determine the type of segment formation used */
+  _segment_formation = track_generator->getSegmentFormation();
+}
+
+
+/**
+ * @brief Destructor for TraverseTracks
+ */
+TraverseTracks::~TraverseTracks() {
+}
+
+
+/**
+ * @breif Loops over Tracks, applying the provided kernels to all segments and
+ *        the functionality described in onTrack(...) to all Tracks.
+ * @details The segment formation method imported from the TrackGenerator
+ *          during construction is used to redirect to the appropriate looping
+ *          scheme. If kernels are provided (not NULL) then they are deleted at
+ *          the end of the looping scheme.
+ * @param kernels MOCKernels to apply to all segments
+ */
+void TraverseTracks::loopOverTracks(MOCKernel** kernels) {
+
+  switch (_segment_formation) {
```

Comment:
Not really necessary since ``_segment_formation`` is an enum type that we define and all options should be supported.